maestro-cli/src/main/java/maestro/cli/cloud/CloudInteractor.kt

```diff
@@ -242,9 +260,12 @@ class CloudInteractor(
             )
         )
 
-        val passed = if (upload.status == UploadStatus.Status.CANCELED && failOnCancellation) {
-            false
-        } else upload.status != UploadStatus.Status.ERROR
+        val isCancelled = upload.status == UploadStatus.Status.CANCELED
+        val isFailure = upload.status == UploadStatus.Status.ERROR
+        val containsFailure =
+            upload.flows.find { it.status == UploadStatus.Status.ERROR } != null // status can be cancelled but also contain flow with failure
+
+        val passed = !isFailure && !containsFailure && !(isCancelled && failOnCancellation)
```

Comment:
Could we invert this logic to avoid all the negations? It is pretty hard to read 😅

Reply:
Simplified it a bit for easier readability